src/controller/JCMCommissioner.cpp

```diff
@@ -0,0 +1,430 @@
+                bool foundMatchingFabricIndex = false;
+
+                auto iter = fabrics.begin();
+                while (iter.Next())
+                {
+                    auto & fabricDescriptor = iter.GetValue();
+                    if (fabricDescriptor.fabricIndex != kUndefinedFabricIndex)
+                    {
+                        if (fabricDescriptor.rootPublicKey.size() != Crypto::kP256_PublicKey_Length)
+                        {
+                            ChipLogError(Controller, "JCM: DeviceCommissioner::ParseJFAdministratorInfo - fabric root key size mismatch");
+                            return CHIP_ERROR_KEY_NOT_FOUND;
+                        }
+
+                        mInfo.rootKeySpan = fabricDescriptor.rootPublicKey;
+                        mInfo.adminVendorId = fabricDescriptor.vendorID;
+                        mInfo.adminFabricId = fabricDescriptor.fabricID;
+
+                        if (fabricDescriptor.VIDVerificationStatement.HasValue())
+                        {
+                            ChipLogError(Controller, "JCM: Per-home RCAC are not supported by JF for now!");
+                            return CHIP_ERROR_CANCELLED;
+                        }
+
+                        foundMatchingFabricIndex = true;
+                        ChipLogProgress(Controller, "JCM: Successfully parsed the Administrator Fabric Table");
+                        break;
+                    }
+                }
+
+                if (!foundMatchingFabricIndex)
+                {
+                    return CHIP_ERROR_NOT_FOUND;
+                }
+                return CHIP_NO_ERROR;
+            }
+            case NOCs::Id: {
+                NOCs::TypeInfo::DecodableType nocs;
+                ReturnErrorOnFailure(attributeCache->Get<NOCs::TypeInfo>(path, nocs));
+
+                auto iter = nocs.begin();
+                while (iter.Next())
+                {
+                    auto & nocStruct = iter.GetValue();
+
+                    if (nocStruct.fabricIndex == mInfo.adminFabricIndex)
+                    {
+                        ReturnErrorOnFailure(AllocateMemoryAndCopySpan(mInfo.adminNOC, nocStruct.noc));
+
+                        if (!nocStruct.icac.IsNull())
+                        {
+                            auto icac = nocStruct.icac.Value();
+                            ReturnErrorOnFailure(AllocateMemoryAndCopySpan(mInfo.adminICAC, icac));
+                        }
+                        else
+                        {
+                            ChipLogError(Controller, "JCM: ICAC not present!");
+                            return CHIP_ERROR_CERT_NOT_FOUND;
+                        }
+                        ChipLogProgress(Controller, "JCM: Successfully parsed the Administrator NOC and ICAC");
+                        break;
+                    }
+                }
+                return CHIP_NO_ERROR;
+            }
+            default:
+                return CHIP_NO_ERROR;
```

Comment:
this return CHIP_NO_ERROR even if no valid attribute if found. please also reuse the validity checks from the existing code.